### ISSUE TITLE
Search does queries twice

### ISSUE THREAD

**Issue by pjotrp** [genenetwork/genenetwork2#170]
According to my logger when I do a search the same SQL is run from two different functions (retrieve_info and get_trait_info) for every row in the table:

```sql
INFO:base.trait:.sql: retrieve_info:
                            SELECT
                                    ProbeSetXRef.Locus, ProbeSetXRef.LRS, ProbeSetXRef.pValue, ProbeSetXRef.mean, ProbeSetXRef.additive
                            FROM
                                    ProbeSetXRef, ProbeSet
                            WHERE
                                    ProbeSetXRef.ProbeSetId = ProbeSet.Id AND
                                    ProbeSet.Name = "1443823_s_at" AND
                                    ProbeSetXRef.ProbeSetFreezeId =112

INFO:base.trait:.sql: retrieve_info:
                                select Geno.Chr, Geno.Mb from Geno, Species
                                where Species.Name = 'mouse' and
                                Geno.Name = 'NES13033186' and
                                Geno.SpeciesId = Species.Id

INFO:base.data_set:.sql: get_trait_info: select ProbeSetXRef.mean from ProbeSetXRef, ProbeSet
                where ProbeSetXRef.ProbeSetFreezeId = 112 and
                ProbeSet.Id = ProbeSetXRef.ProbeSetId and
                ProbeSet.Name = '1443823_s_at'

INFO:base.data_set:.sql: get_trait_info:
                    select Geno.Chr, Geno.Mb from Geno, Species
                    where Species.Name = 'mouse' and
                        Geno.Name = 'NES13033186' and
                        Geno.SpeciesId = Species.Id
```

**Comment by pjotrp** [comment]
Merge my staging tree. Some new parameters have been introduced:

```python
LOG_SQLALCHEMY  = 'False'
LOG_BENCH       = True      # Log bench marks
USE_REDIS       = True      # REDIS caching (note that redis will be phased out)
USE_GN_SERVER   = 'False'   # Use GN_SERVER SQL calls
```

see the examples in ./etc/default_settings.py

I now run GN2 with

```sh
env LOG_LEVEL=DEBUG USE_REDIS=False LOG_BENCH=True USE_LOG_SQL=True LOG_LEVEL_DEBUG=1 ./bin/genenetwork2 ~/default_settings.py
```
to override logging behaviour.

**Comment by pjotrp** [comment]
Staging is running on http://test-gn2.genenetwork.org/ - I have to fix the CSS

**Comment by zsloan** [comment]
Just to clarify, do you want me to merge your code on the staging branch
and then merge that to production if it seems to be working okay?

Nice catch with that query; I can't seem to find any reason for the
"self.dataset.get_trait_info(self.trait_list, species)" line or the
"get_info" function in trait.py to exist (the latter is called in
collect.py but doesn't seem to change anything if it is removed.

**Comment by pjotrp** [comment]
Yes, please merge staging.

Another one is that this query is invoked for every row. It can actually easily be run in the main search call since all rows are fetched and the two tables are in that query:

```sql
INFO:base.trait:.sql: retrieve_info: 
                            SELECT
                                    ProbeSetXRef.Locus, ProbeSetXRef.LRS, ProbeSetXRef.pValue, ProbeSetXRef.mean, ProbeSetXRef.additive
                            FROM
                                    ProbeSetXRef, ProbeSet
                            WHERE
                                    ProbeSetXRef.ProbeSetId = ProbeSet.Id AND
                                    ProbeSet.Name = "1426659_a_at" AND
                                    ProbeSetXRef.ProbeSetFreezeId =112
```

No need to fix now. Just leave the issue open.

**Comment by pjotrp** [comment]
Also the main query - are we really using all these fields to display in search_results?

```sql
INFO:base.trait:.sql: retrieve_info: 
                    SELECT ProbeSet.name, ProbeSet.symbol, ProbeSet.description, ProbeSet.probe_target_description, ProbeSet.chr, ProbeSet.mb, ProbeSet.alias, ProbeSet.geneid, ProbeSet.genbankid, ProbeSet.unigeneid, ProbeSet.omim, ProbeSet.refseq_transcriptid, ProbeSet.blatseq, ProbeSet.targetseq, ProbeSet.chipid, ProbeSet.comments, ProbeSet.strand_probe, ProbeSet.strand_gene, ProbeSet.probe_set_target_region, ProbeSet.probe_set_specificity, ProbeSet.probe_set_blat_score, ProbeSet.probe_set_blat_mb_start, ProbeSet.probe_set_blat_mb_end, ProbeSet.probe_set_strand, ProbeSet.probe_set_note_by_rw, ProbeSet.flag
                    FROM ProbeSet, ProbeSetFreeze, ProbeSetXRef
                    WHERE
                            ProbeSetXRef.ProbeSetFreezeId = ProbeSetFreeze.Id AND
                            ProbeSetXRef.ProbeSetId = ProbeSet.Id AND
                            ProbeSetFreeze.Name = 'HC_M2_0606_P' AND
                            ProbeSet.Name = '1455792_x_at'
```

**Comment by zsloan** [comment]
I pulled your changes and things seem to be working fine after I edited the
settings file.

As for the retrieve_info query...I think most, if not all, of those fields
are used somewhere, but not all are used in the search results. Most are
used in the trait page. For example the blat is used when generating the
link for the "Verify" button on the trait plage. Does it actually slow down
things much to just select more columns from the same table?
